src/Core/Security/AdminLoginFormThrottlingConfiguration.php

```diff
@@ -0,0 +1,241 @@
+    {
+        $enabled = $this->getBoolean('PS_SECURITY_ADMIN_LOGIN_THROTTLING_ENABLED', true);
+        $maxAttempts = $this->getPositiveInt('PS_SECURITY_ADMIN_LOGIN_THROTTLING_MAX_ATTEMPTS', self::DEFAULT_MAX_ATTEMPTS);
+        $intervalMinutes = $this->getPositiveInt('PS_SECURITY_ADMIN_LOGIN_THROTTLING_INTERVAL', self::DEFAULT_INTERVAL_MINUTES);
+        $storageService = $this->getString('PS_SECURITY_ADMIN_LOGIN_THROTTLING_STORAGE', '');
+
+        return [
+            'login_throttling_enabled' => $enabled,
+            'login_throttling_max_attempts' => $maxAttempts,
+            'login_throttling_interval' => $intervalMinutes,
+            'login_throttling_storage' => $storageService,
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateConfiguration(array $configuration): array
+    {
+        $errors = $this->getValidationErrors($configuration);
+
+        if (!empty($errors)) {
+            return $errors;
+        }
+
+        $this->configuration->set('PS_SECURITY_ADMIN_LOGIN_THROTTLING_ENABLED', (int) (bool) $configuration['login_throttling_enabled']);
+        $this->configuration->set('PS_SECURITY_ADMIN_LOGIN_THROTTLING_MAX_ATTEMPTS', (int) $configuration['login_throttling_max_attempts']);
+        $this->configuration->set('PS_SECURITY_ADMIN_LOGIN_THROTTLING_INTERVAL', (int) $configuration['login_throttling_interval']);
+        $this->configuration->set('PS_SECURITY_ADMIN_LOGIN_THROTTLING_STORAGE', trim((string) $configuration['login_throttling_storage']));
+
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validateConfiguration(array $configuration): bool
+    {
+        return [] === $this->getValidationErrors($configuration);
+    }
+
+    public function getFirewallConfiguration(): array|bool
+    {
+        $configuration = $this->getConfiguration();
+        $enabled = (bool) $configuration['login_throttling_enabled'];
+        $maxAttempts = (int) $configuration['login_throttling_max_attempts'];
+        $intervalMinutes = (int) $configuration['login_throttling_interval'];
+        $storageService = (string) $configuration['login_throttling_storage'];
+
+        if (!$enabled) {
+            return false;
+        }
+
+        $intervalString = $intervalMinutes === 1 ? '1 minute' : sprintf('%d minutes', $intervalMinutes);
+
+        $configuration = [
+            'max_attempts' => $maxAttempts,
+            'interval' => $intervalString,
+        ];
+
+        if ($storageService !== '') {
+            $configuration['storage_service'] = $storageService;
+        }
+
+        return $configuration;
+    }
+
+    private function getBoolean(string $key, bool $default): bool
+    {
+        $value = $this->configuration->get($key);
+
+        if ($value === false) {
+            return $default;
+        }
+
+        return (bool) $value;
+    }
+
+    private function getPositiveInt(string $key, int $default): int
```

Comment:
Do these methods really need to be here and belong here? 🤔 I don't think that a configuration provider should implement an integer parser.

Also, can't you just do (bool) or (string) cast here in these methods? I think it would bring the same results no?

Reply:
that’s one of the points I was planning to refactor before moving the PR out of draft. 👍